### DISCUSSION TITLE
Support running in Cloudflare Worker environment

### DISCUSSION THREAD

**Discussion by MitchLillie** [jsforce/jsforce#1500] [Ideas]
When trying to run jsforce in a Cloudflare worker, it incorrectly determines that it is in a browser environment:

```javascript
const conn = new jsforce.Connection({
    loginUrl: 'https://my.url.here.salesforce.com',
});

addEventListener('fetch', event => {
    event.respondWith(updateSalesForce(event.request));
});

function updateSalesForce() {
    return new Promise((res, rej) => {
        conn.login('user@email.com', 'password', (err) => {
            if (err) {
                console.error(`Error logging into SF: ${err}`);
                rej();
            }
            console.log(`Access Token: ${conn.accessToken}`);
            res();
        });
    });
}
```

When running in a Cloudflare Worker, the above throws an error, `ReferenceError: XMLHttpRequest is not defined`. Wherever jsforce is detecting browser vs Node.js, it is failing to detect this environment correctly. The same code works perfectly in Node.js.

**Comment by thiagomagro** [comment]
Im facing this issue too with SvelteKit.

**Comment by jclark-dot-org** [comment]
Cloudflare Worker's Node compat layer doesn't support Node's HTTP either ("not yet supported" per the [compat page](https://developers.cloudflare.com/workers/runtime-apis/nodejs/));  Switching from XMLHttpRequest to the fetch API would solve the issue and probably provide some future-proofing.

**Comment by Damecek** [comment]
Same issue here running in Chrome Extension `service_worker`
